pages/api/replicache-pull.ts

```diff
@@ -1,51 +1,109 @@
 import * as t from "io-ts";
 import type { NextApiRequest, NextApiResponse } from "next";
-import { ExecuteStatementCommandOutput } from "@aws-sdk/client-rds-data";
+import { ExecuteStatementCommandOutput, Field } from "@aws-sdk/client-rds-data";
 import { transact } from "../../backend/rds";
-import { getLastMutationID } from "../../backend/data";
+import { getCookieVersion, getLastMutationID } from "../../backend/data";
 import { must } from "../../backend/decode";
 
 export default async (req: NextApiRequest, res: NextApiResponse) => {
   console.log(`Processing pull`, req.body);
 
   const pull = must(pullRequest.decode(req.body));
+  let cookie = pull.baseStateID == "" ? 0 : parseInt(pull.baseStateID);
 
   console.time(`Reading all Shapes...`);
   let entries;
   let lastMutationID = 0;
 
   await transact(async (executor) => {
-    entries = (await executor("SELECT * FROM Shape")) ?? [];
-    lastMutationID = await getLastMutationID(executor, pull.clientID);
+    [entries, lastMutationID, cookie] = await Promise.all([
+      executor("SELECT * FROM Shape WHERE Version > :version", {
+        version: { longValue: cookie },
+      }),
+      getLastMutationID(executor, pull.clientID),
+      getCookieVersion(executor),
+    ]);
   });
+  console.log({ lastMutationID });
   console.timeEnd(`Reading all Shapes...`);
 
   // Grump. Typescript seems to not understand that the argument to transact()
   // is guaranteed to have been called before transact() exits.
   entries = (entries as any) as ExecuteStatementCommandOutput;
 
+  const resp: PullResponse = {
+    lastMutationID,
```

Comment:
if the client lmid is > the lastmutationid we have here we should probably send a nop. as mentioned above, we don't have that signal so send an empty patch with the lmid and cookie supplied by the client. (and probably log)